astropy/coordinates/baseframe.py

```diff
@@ -13,27 +13,22 @@
 import copy
 import inspect
 from collections import namedtuple, OrderedDict, defaultdict
+import warnings
 
 # Dependencies
 import numpy as np
 
 # Project
 from ..utils.compat.misc import override__dir__
 from ..utils.decorators import lazyproperty
+from ..utils.exceptions import AstropyWarning
 from ..extern import six
 from ..extern.six.moves import zip
 from .. import units as u
 from ..utils import (OrderedDescriptorContainer, ShapedLikeNDArray,
                      check_broadcast)
-from ..utils.misc import isiterable
 from .transformations import TransformGraph
-from .representation import (BaseRepresentation, BaseDifferential,
```

Comment:
I mind not much in the built-in cases, but here isn't it better to use the explicit import? That way there's more autochecking on whether you import what you need. It's also marginally faster code....  But no big deal; fine to leave as you have it now too.